### ISSUE TITLE
Symlic resolve problem?

### ISSUE THREAD

**Issue by kosch** [marschall/memoryfilesystem#60]
When trying to resolve symlink between two directories with toRealPath(), I noticed some problems. Source below explains what I expect:

```java
import com.github.marschall.memoryfilesystem.MemoryFileSystemBuilder;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.nio.file.FileSystem;
import java.nio.file.Files;
import java.nio.file.Path;

import static org.junit.Assert.*;

public class SymlinkTest {

    private FileSystem fileSystem;
    private Path targetPath;

    @Before
    public void before() throws Exception {
        fileSystem = MemoryFileSystemBuilder.newEmpty().build("test1");

        // lets prepare the filesystem content...

        targetPath = fileSystem.getPath("target");
        Files.createDirectory(targetPath);

        // now create a link from /src -> /target
        Path srcPath = fileSystem.getPath("src");
        Files.createSymbolicLink(srcPath, targetPath);

        // lets check a bit here
        assertEquals("/target", targetPath.toAbsolutePath().toString());
        assertEquals("/src", srcPath.toAbsolutePath().toString());

        Path srcPath2 = fileSystem.getPath("src");
        assertTrue(Files.isSymbolicLink(srcPath2));
        assertTrue(Files.isDirectory(srcPath2));
        assertEquals(srcPath, srcPath2);

        // so far so good
    }

    @After
    public void after() throws Exception {
        fileSystem.close();
    }

    @Test
    public void realPathEquals() throws Exception {
        Path realPath = fileSystem.getPath("src").toRealPath();
        assertEquals(targetPath, realPath);
    }

    @Test
    public void realPathDirectory() throws Exception {
        Path realPath = fileSystem.getPath("src").toRealPath();
        assertTrue(Files.isDirectory(realPath));
    }

    @Test
    public void realPathSymlink() throws Exception {
        Path realPath = fileSystem.getPath("src").toRealPath();
        assertFalse(Files.isSymbolicLink(realPath));
    }

    @Test
    public void realPathToString() throws Exception {
        Path realPath = fileSystem.getPath("src").toRealPath();
        assertEquals("/target", realPath.toString());
    }

}
```

**Comment by marschall** [comment]
`#toRealPath()` was indeed buggy and created invalid path objects, that was why most of the tests failed. After fixing `#realPathEquals()` was still failing because `#toRealPath()` returns an absolute path whereas `targetPath` is a relative path. Writing the test like this:

```java
  @Test
  public void realPathEquals() throws IOException {
    Path realPath = this.fileSystem.getPath("src").toRealPath();
    assertEquals(this.targetPath.toAbsolutePath(), realPath);
  }
```

makes it pass.

**Comment by marschall** [comment]
I just released 0.7.3, it should be in Maven Central soonish (a few hours).
